core/domain/state_domain.py

```diff
@@ -1461,6 +1466,17 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                     'Found a duplicate content id %s' % solution_content_id)
             content_id_list.append(solution_content_id)
 
+        if not isinstance(self.solicit_answer_details, bool):
+            raise utils.ValidationError(
+                'Expected solicit_answer_details to be a boolean, ' +
+                'received %s' % self.solicit_answer_details)
+        if self.solicit_answer_details:
+            if self.interaction.id != 'TextInput':
```

Comment:
Ah, let's not hardcode this. I was thinking to add 'can_solicit_answer_details = False` inside `class BaseInteraction` (or something like this) and make it to true inside `extensions/interactions/TextInput/TextInput.py`.

Reply:
not sure about the naming on `can_solicit_answer_details`*

Reply:
Hi Sandeep, also one option is I can make list named CAN_SOLICIT_ANSWER_DETAILS in feconf.py, which will include the interaction ids which can have this value as true. What do you suggest?

Reply:
If your functionality depends on interaction type then I think we should use the existing interaction configuration structure.

Reply:
Incidentally the feature you are building should not depend on a specific interaction type @anubhavsinha98.

It is OK to exclude a subset of interactions which obviously don't need it (e.g. Continue, EndExploration). But the answer details feature should be implemented as a general thing, not just specialized to TextInput.

Reply:
Yeah @seanlip, you are right. This feature does not depend on a specific interaction type. As currently, I would be testing the feature for TextInput interaction only as decided in my project, but yeah the whole system should be generalized. So that extending this feature is easy.

So to do that I was thinking to make a list variable in feconf file:
```CAN_SOLICIT_ANSWER_DETAILS = ['TextInput']```
And later on, if we want to extend this feature to any of the interaction types we just need to mention its id in the list variable, and then the creator will be able to make ```solicit_answer_details``` true for those interaction also.
Does this idea seems fine?

Reply:
There's no reason to exclude any nontrivial interaction ID, even initially. I would do them all at once in this PR.

Reply:
Okay @seanlip, I will include all the nontrivial interaction IDs in it, by the way does the idea to make a list in feconf seems fine?

Reply:
The feconf list should be a blacklist. I.e. it contains the list of IDs that are not used -- Continue, EndExploration, etc. It should generally not be modified, since what you implement ought to be sufficiently general to work for any new interaction that is subsequently added.

Reply:
Okay @seanlip, thanks. Seems fine :)